src/compiler/checker.ts

```diff
@@ -56,6 +56,10 @@ namespace ts {
 
         const undefinedSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.Transient, "undefined");
         undefinedSymbol.declarations = [];
+        const NaNSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.TypeAlias | SymbolFlags.Transient, (NaN).toString());

```

Comment:
`TypeAlias`? This seems like an inappropriate way to piggy-back on an existing flag.